drivers/interrupt_controller/intc_plic.c

```diff
@@ -93,18 +95,60 @@ static inline uint32_t get_plic_enabled_size(const struct device *dev)
 	return local_irq_to_reg_index(config->num_irqs) + 1;
 }
 
+static inline uint16_t get_first_context(uint16_t hartid)
+{
+	return hartid == 0 ? 0 : (hartid * 2) - 1;
+}
+
+static inline mem_addr_t get_context_en_addr(const struct device *dev, uint16_t cpu_num)
+{
+	const struct plic_config *config = dev->config;
+	int hartid;
```

Comment:
this is getting casted into different type at different places, should probably just fix it to `uint32_t` (the type of `arch_proc_id()`)

Reply:
@ycsin, thanks, makes sense: will do

Reply:
fixed

Reply:
this is still `int` in the latest revision

Reply:
@ycin ah, my bad - I will fix this now

Reply:
@ycsin done